packages/material-ui/src/internal/SwitchBase.test.js

```diff
@@ -1,445 +1,358 @@
-      wrapper
-        .find('input')
-        .instance()
-        .click();
-      wrapper.update();
-      assertIsChecked(wrapper);
-    });
+    getByRole('checkbox').click();
 
-    it('should uncheck the checkbox', () => {
-      wrapper
-        .find('input')
-        .instance()
-        .click();
-      wrapper
-        .find('input')
-        .instance()
-        .click();
-      wrapper.update();
-      assertIsNotChecked(wrapper);
-    });
-  });
+    expect(container.firstChild).not.to.have.class(classes.checked);
+    expect(getByRole('checkbox')).to.have.property('checked', false);
+    expect(getByTestId('unchecked-icon')).to.be.ok;
 
-  describe('prop: icon', () => {
-    it('should render an Icon', () => {
-      const wrapper = mount(<SwitchBase {...defaultProps} icon={<Icon>heart</Icon>} />);
-      assert.strictEqual(wrapper.contains(Icon), true);
-    });
+    getByRole('checkbox').click();
+
+    expect(container.firstChild).to.have.class(classes.checked);
+    expect(getByRole('checkbox')).to.have.property('checked', true);
+    expect(getByTestId('checked-icon')).to.be.ok;
   });
 
   describe('handleInputChange()', () => {
-    const eventMock = 'something-to-match';
-    const event = {
-      target: {
-        checked: false,
-      },
-      eventMock,
-    };
-
-    it('should call onChange exactly once with event', () => {
-      const handleChange = spy();
-      const wrapper = mount(
-        <SwitchBaseNaked {...defaultProps} classes={{}} onChange={handleChange} />,
+    it('should call onChange when uncontrolled', () => {
+      const handleChange = spy(event => event.target.checked);
+      const { getByRole } = render(
+        <SwitchBase
+          icon="unchecked"
+          checkedIcon="checked"
+          type="checkbox"
+          onChange={handleChange}
+        />,
       );
 
-      const input = wrapper.find('input');
-      input.simulate('change', event);
+      getByRole('checkbox').click();
 
-      assert.strictEqual(handleChange.callCount, 1);
-      assert.strictEqual(handleChange.calledWithMatch(event), true);
-
-      handleChange.resetHistory();
-    });
-
-    describe('controlled', () => {
-      it('should call onChange once', () => {
-        const checked = true;
-        const handleChange = spy();
-        const wrapper = mount(
-          <SwitchBaseNaked
-            {...defaultProps}
-            classes={{}}
-            checked={checked}
-            onChange={handleChange}
-          />,
-        );
-        const input = wrapper.find('input');
-        input.simulate('change', event);
-
-        assert.strictEqual(handleChange.callCount, 1);
-        assert.strictEqual(
-          handleChange.calledWithMatch(event, !checked),
-          true,
-          'call onChange with event and !props.checked',
-        );
-      });
+      expect(handleChange.callCount).to.equal(1);
+      // event.target.check is true
+      expect(handleChange.firstCall.returnValue).to.be.true;
```

Comment:
What's the advantage over using the generic equal?

```suggestion
      expect(handleChange.firstCall.returnValue).to.equal(true);
```

For reference, I can't find this API in the Jest documentation: https://jestjs.io/.

Reply:
@oliviertassinari 
https://jestjs.io/docs/en/expect.html#toequalvalue
https://jestjs.io/docs/en/expect.html#tobevalue

Reply:
@merceyz Thanks for the links. Do you know if there is a `.toBeTrue()` like API?

Reply:
https://jestjs.io/docs/en/expect.html#tobetruthy

That's the closest you'll get

Reply:
@eps1lon What do you think?

Reply:
I'm not following what the issue is? This appeared to be a question about jest which was answered

Reply:
My proposal is to replace the usages of `to.be.true` and `to.be.false` with the generic `to.be.equal` API. I have used Jest's API as a benchmark.